src/core/meshcop/secure_transport.hpp

```diff
@@ -435,6 +435,19 @@ class SecureTransport : private NonCopyable
 #endif // defined(MBEDTLS_BASE64_C) && defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
 
 #if defined(MBEDTLS_SSL_KEEP_PEER_CERTIFICATE)
+        /**
+         * Returns the DER encoded peer x509 certificate.
+         *
+         * @param[out]  aPeerCert        A pointer to the DER encoded certificate buffer.
+         * @param[out]  aCertLength      The length of the DER encoded peer certificate.
+         * @param[in]   aCertBufferSize  The buffer size of aPeerCert.
+         *
+         * @retval kErrorInvalidState   Not connected yet.
+         * @retval kErrorNone           Successfully get the peer certificate.
+         * @retval kErrorNoBufs         Can't allocate memory for certificate.
+         */
+        Error GetPeerCertificateDer(unsigned char *aPeerCert, size_t *aCertLength, size_t aCertBufferSize);
```

Comment:
Suggest using reference instead of pointer for the output variable? `size_t &aCertLength`?
(unless it being `nullptr` has a a use)

Reply:
This is for API consistency pleases see `GetPeerCertificateBase64`.

Reply:
@abtink I can change both of those API calls but I would prefer to do it in separate PR.

Reply:
Sounds good. Let's changing it later.